File: 10.0.3.020/examples/vmlf/source/vdsincos.f
!*******************************************************************************
!                              INTEL CONFIDENTIAL
!  Copyright(C) 2001-2008 Intel Corporation. All Rights Reserved.
!  The source code contained  or  described herein and all documents related to
!  the source code ("Material") are owned by Intel Corporation or its suppliers
!  or licensors.  Title to the  Material remains with  Intel Corporation or its
!  suppliers and licensors. The Material contains trade secrets and proprietary
!  and  confidential  information of  Intel or its suppliers and licensors. The
!  Material  is  protected  by  worldwide  copyright  and trade secret laws and
!  treaty  provisions. No part of the Material may be used, copied, reproduced,
!  modified, published, uploaded, posted, transmitted, distributed or disclosed
!  in any way without Intel's prior express written permission.
!  No license  under any  patent, copyright, trade secret or other intellectual
!  property right is granted to or conferred upon you by disclosure or delivery
!  of the Materials,  either expressly, by implication, inducement, estoppel or
!  otherwise.  Any  license  under  such  intellectual property  rights must be
!  express and approved by Intel in writing.
!
!*******************************************************************************
!  Content:
!    vdSinCos  Example Program Text
!*******************************************************************************

      program MKL_VML_TEST

      include "mkl_vml.fi"
      include "func_interv.fi"

      real(kind=8) dA(VEC_LEN)
      real(kind=8) dB1(VEC_LEN)
      real(kind=8) dB2(VEC_LEN)
      real(kind=8) dB3(VEC_LEN)
      real(kind=8) dB4(VEC_LEN)
      real(kind=8) CurRMS1,MaxRMS1
      real(kind=8) CurRMS2,MaxRMS2
      real(kind=8) MaxRMS
      integer i

      MaxRMS=0.0
      MaxRMS1=0.0
      MaxRMS2=0.0
      do i=1,VEC_LEN
            dA(i)=VML_DSINCOS_BEG+((VML_DSINCOS_END-VML_DSINCOS_BEG)*i) &
     &            /VEC_LEN
            dB1(i)=0.0
            dB2(i)=0.0
            dB3(i)=sin(dA(i))
            dB4(i)=cos(dA(i))
      end do

      call VDSINCOS(VEC_LEN,dA,dB1,dB2)

      print *,"vdSinCos test/example program"
      print *,""
      print *,""
      print *,"           Argument                     vdSinCos      ", &
     &        "                SinCos"
      print *,"======================================================", &
     &        "========================"
      do i=1,VEC_LEN
            print 10,dA(i),dB1(i),dB2(i),dB3(i),dB4(i)
            CurRMS1=abs((dB1(i)-dB3(i))/(0.5*(dB1(i)+dB3(i))))
            CurRMS2=abs((dB2(i)-dB4(i))/(0.5*(dB2(i)+dB4(i))))
            if(MaxRMS1<CurRMS1) MaxRMS1=CurRMS1
            if(MaxRMS2<CurRMS2) MaxRMS2=CurRMS2
      end do

      if(MaxRMS2<MaxRMS1) MaxRMS=MaxRMS1
      if(MaxRMS2>MaxRMS1) MaxRMS=MaxRMS2
      print *,""
      if(MaxRMS>=DEPS) then
            print 11,"Error! Relative accuracy is ",MaxRMS
            stop 1
      else
            print 11,"Relative accuracy is ",MaxRMS
      endif

10    format(F22.14,F20.6,F10.6,F13.6,F10.6)
11    format(A,F25.14)

      end

File: 10.0.3.020/examples/java/examples/vdsincos.java
/*******************************************************************************
!                             INTEL CONFIDENTIAL
!   Copyright(C) 2007-2008 Intel Corporation. All Rights Reserved.
!   The source code contained  or  described herein and all documents related to
!   the source code ("Material") are owned by Intel Corporation or its suppliers
!   or licensors.  Title to the  Material remains with  Intel Corporation or its
!   suppliers and licensors. The Material contains trade secrets and proprietary
!   and  confidential  information of  Intel or its suppliers and licensors. The
!   Material  is  protected  by  worldwide  copyright  and trade secret laws and
!   treaty  provisions. No part of the Material may be used, copied, reproduced,
!   modified, published, uploaded, posted, transmitted, distributed or disclosed
!   in any way without Intel's prior express written permission.
!   No license  under any  patent, copyright, trade secret or other intellectual
!   property right is granted to or conferred upon you by disclosure or delivery
!   of the Materials,  either expressly, by implication, inducement, estoppel or
!   otherwise.  Any  license  under  such  intellectual property  rights must be
!   express and approved by Intel in writing.
!******************************************************************************/

import com.intel.mkl.*;

public final class vdsincos {
    /**
     * Incarnation prohibited.
     */
    private vdsincos() {}

    /**
     * Individual name of this test.
     */
    private final static String test_name = "vdsincos";

    /**
     * No command-line options.
     */
    public static void main(String[] args) {
        boolean fault;
        try {
            fault = test();
        } catch (Exception statusException) {
            fault = true;
            statusException.printStackTrace(System.out);
            System.out.println("\nTEST FAILED");
            System.exit(1);
        }
        System.out.println("\nTEST PASSED");
    }

    /**
     * Demonstrate the VML functions and test them.
     * @return true - If the test has failed.
     */
    public static boolean test()
    {
        // Parameters
        final double DEPS = 0.1;
        final int VEC_LEN = 11;
        final double __DSINCOS_BEG = -10000.0;
        final double __DSINCOS_END = 10000.0;

        double[] dA1 = new double[VEC_LEN];
        double[] dB1 = new double[VEC_LEN];
        double[] dB2 = new double[VEC_LEN];
        double[] dB3 = new double[VEC_LEN];
        double[] dB4 = new double[VEC_LEN];

        int i=0,vec_len=VEC_LEN;
        double CurRMS,MaxRMSsin=0.0,MaxRMScos=0.0;

        for(i=0;i<vec_len;i++) {
                dA1[i]=(double)(__DSINCOS_BEG+((__DSINCOS_END-__DSINCOS_BEG)*i)/vec_len);
                dB1[i]=0.0;
                dB2[i]=0.0;
                dB3[i]=Math.sin(dA1[i]);
                dB4[i]=Math.cos(dA1[i]);
        }

        VML.vdSinCos(vec_len,dA1,dB1,dB2);

        System.out.println("vdSinCos test/example program\n");
        System.out.println("        Argument         vdSinCos:   Sin        Cos           Sin        Cos");
        System.out.println("===============================================================================");
        for(i=0;i<vec_len;i++) {
                System.out.println("" + dA1[i] + " " + dB1[i] + " " + dB2[i] + " " + dB3[i] + " " + dB4[i] + "");
                CurRMS=(dB1[i]-dB3[i])/(0.5*(dB1[i]+dB3[i]));
                if(MaxRMSsin<CurRMS) MaxRMSsin=CurRMS;
                CurRMS=(dB2[i]-dB4[i])/(0.5*(dB2[i]+dB4[i]));
                if(MaxRMScos<CurRMS) MaxRMScos=CurRMS;
        }
        System.out.println("");
        if(MaxRMSsin>=DEPS) {
                System.out.println("Error! Relative accuracy (Math.sin) is " + MaxRMSsin + "");
        }
        else {
                System.out.println("Relative accuracy (Math.sin) is " + MaxRMSsin + "");
        }
        if(MaxRMScos>=DEPS) {
                System.out.println("Error! Relative accuracy (Math.cos) is " + MaxRMScos + "");
        }
        else {
                System.out.println("Relative accuracy (Math.cos) is " + MaxRMScos + "");
        }

        return false;
    }
}
